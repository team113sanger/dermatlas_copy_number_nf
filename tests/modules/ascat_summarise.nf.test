nextflow_process {

    name "Test Process SUMMARISE_ASCAT_ESTIMATES"
    script "modules/ascat.nf"
    process "SUMMARISE_ASCAT_ESTIMATES"

    test("Should run without failures") {

        when {
            params {
                load("$baseDir/tests/testdata/test_params.json")
            }
            process {
            
                """
                input[0] = tuple([analysis_type: "test"],
                     ["${baseDir}/tests/testdata/ASCAT_estimates_PD62468a_PD62468b.tsv",
                      "${baseDir}/tests/testdata/ASCAT_estimates_PD54403a_PD54403b.tsv"])
                """
            }
        }

        then {
            assert process.success
        }

    }

}

