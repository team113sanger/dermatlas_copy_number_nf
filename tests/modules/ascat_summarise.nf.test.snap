{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "ascat_stats.tsv:md5,a102df90ee75b4a9473c27aa75767c50"
                ],
                "1": [
                    "ascat_low_qual.list:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "sample_purity_ploidy.tsv:md5,80485b1f3983333638273f56d98f317f"
                ],
                "ascat_sstats": [
                    "ascat_stats.tsv:md5,a102df90ee75b4a9473c27aa75767c50"
                ],
                "low_quality": [
                    "ascat_low_qual.list:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "purity": [
                    "sample_purity_ploidy.tsv:md5,80485b1f3983333638273f56d98f317f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-29T14:55:24.315469"
    }
}